auth/state.go

```diff
@@ -18,7 +19,7 @@ const (
 // OauthStateReference represents a oauth state reference
 type OauthStateReference struct {
 	gormsupport.Lifecycle
-	ID       uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"`
+	ID       string `sql:"type:uuid default CAST(uuid_generate_v4() AS varchar(50))" gorm:"primary_key"`
```

Comment:
Though this does not create any problem, but I just need to make sure that this is what we want or this is the right way.